include/zephyr/kernel.h

```diff
@@ -1488,13 +1488,16 @@ const char *k_thread_state_str(k_tid_t thread_id, char *buf, size_t buf_size);
  * This macro generates a timeout delay that represents an expiration
  * at the absolute uptime value specified, in system ticks.  That is, the
  * timeout will expire immediately after the system uptime reaches the
- * specified tick count.
+ * specified tick count. INT64_MAX is not a valid argument and will
+ * be changed to (INT64MAX-1), because we use negative values in
+ * k_ticks_t to represent absolute timeouts but -1 is reserved for
+ * K_TIMEOUT_FOREVER.
```

Comment:
In documentation, it's usually clearer just to document what the allowed range is vs. going on a digression about implementation details.  :)

Reply:
I'm on board with that.

I had to catch myself for a second because I was mixing up timeouts and ticks.

Timeouts can optionally be 32-bit which makes things spicy 🌶️